net/net.go

```diff
@@ -1330,25 +1306,57 @@ func (n *net) updateRecordsFromLog(tid thread.ID, lid peer.ID) {
 		return
 	}
 
-	// Get log records for this new log
-	recs, err := n.server.getRecords(
-		n.ctx,
-		tid,
-		lid,
-		map[peer.ID]cid.Cid{lid: cid.Undef},
-		MaxPullLimit)
+	// TODO after protocol change request only new log
```

Comment:
So, are you wanting to follow this up with a protocol change ASAP?

Reply:
Well, as we previously discussed it, protocol change requires getting rid of record integrity check, and you've suggested that the next version of threads protocol should be issued. It'd be great, but not urgent now, as current PR already significantly reduces number of irrelevant records returned by the GetRecords request.

Reply:
Okay, SGTM. I have some other changes that will be nice to get in before a protocol change. So, will be nice to wait for those.